src/state/reducer.ts

```diff
@@ -1,27 +1,30 @@
-const onInitialise = (state, action) => ({
+import {KindeUser, State} from './types';
+
+type Action = {type: 'INITIALISED'; user: KindeUser} | {type: 'ERROR', error: string};
+const onInitialise = (state: State, action: Pick<State, 'user'>): State => ({
   ...state,
-  isAuthenticated: !!action.user,
+  isAuthenticated: Boolean(action.user),
   user: action.user,
   isLoading: false,
   error: undefined
 });
 
-const onComplete = (state, action) =>
-  state.user && state.user.updated_at === action.user && action.user.updated_at
+const onComplete = (state: State, action: Partial<State>): State =>
+  state.user?.updated_at === action.user?.updated_at
     ? state
     : {
         ...state,
         isAuthenticated: !!action.user,
         user: action.user
       };
```

Comment:
I've realised `onComplete` actually isn't used anywhere so this whole function can be removed